batch/batch/driver/scheduler.py

```diff
@@ -190,12 +195,22 @@ def allocate_cores(user, mark):
                 if i < len(self.inst_pool.healthy_instances_by_free_cores):
                     instance = self.inst_pool.healthy_instances_by_free_cores[i]
                     assert record['cores_mcpu'] <= instance.free_cores_mcpu
-                    log.info(f'scheduling job {id} on {instance}')
-                    try:
-                        await schedule_job(self.app, record, instance)
-                    except Exception:
-                        log.exception(f'while scheduling job {id} on {instance}')
+                    instance.adjust_free_cores_in_memory(-record['cores_mcpu'])
                     should_wait = False
                     scheduled_cores_mcpu += record['cores_mcpu']
+                    to_schedule.append((record, instance))
+
+        results = await bounded_gather(*[functools.partial(schedule_job, self.app, record, instance)
```

Comment:
This won't make a difference for one user, but it would be nice if you could start calling `schedule_job` for the first user while scheduling later users.  I want a hybrid of bounded_gather and AsyncWorkerPool, so I can add work to the worker pool, but at the end of the function, call `wait` and wait for all worker pool jobs to complete before resuming.